hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -443,25 +450,26 @@ def _onchange_dates(self):
     def _onchange_timesheets(self):
         self._compute_line_ids()
 
-    @api.onchange("add_line_project_id")
-    def onchange_add_project_id(self):
-        """Load the project to the timesheet sheet"""
-        if self.add_line_project_id:
-            return {
-                "domain": {
-                    "add_line_task_id": [
-                        ("project_id", "=", self.add_line_project_id.id),
-                        ("company_id", "=", self.company_id.id),
-                        ("id", "not in", self.timesheet_ids.mapped("task_id").ids),
+    @api.depends(
+        "add_line_project_id", "company_id", "timesheet_ids", "timesheet_ids.task_id"
+    )
+    def _compute_available_task_ids(self):
+        project_task_obj = self.env["project.task"]
+        for rec in self:
```

Comment:
some nitpick only if you want. instead of `rec` it will be better to have `sheet` as a variable name.